app/validators/application_fulfilment_validator.rb

```diff
@@ -3,10 +3,10 @@ class ApplicationFulfilmentValidator < BaseFulfilmentValidator
 
   # More validations can be added here
   # Errors, when more than one, will maintain the order
-  def perform_validations # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def perform_validations # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity
     errors = []
 
-    unless Passporting::MeansPassporter.new(record).call || evidence_present?
+    unless Passporting::MeansPassporter.new(record).call || evidence_present? || means_record_present?
```

Comment:
In `Passporting::MeansPassporter.new()...` the `.call` method has the following code:

```      means_passport = []
      means_passport << MeansPassportType::ON_NOT_MEANS_TESTED if app_not_means_tested?
      means_passport << MeansPassportType::ON_AGE_UNDER18      if applicant_under18?
      means_passport << MeansPassportType::ON_BENEFIT_CHECK    if benefit_check_passed?

      crime_application.update(means_passport:)
```

I think this is saving the means passporting decision making check? Does this mean skipping means passporting using `means_record_present?` will mean the reason is not persisted to database? Also should `means_record_present?` and `evidence_present?` be first in the if statement to prevent unnecessary object initialisation? Or does the order matter here through side-effects in the other 2 conditions?

Reply:
So the concept is that on validation we rerun the passporter in order to set the passport status based on the values the application has at the point of submission. 
i.e. applicant goes through the form with details that gains them a means passport, then return and change one of those values (e.g. NINO) and dip out and back to the submission page. When they try to save and submit, the passporter is rerun against the new NINO and returns 'No' - so the original means_passport gets scrapped, and the validator checks if there is otherwise evidence (currently only available for evidencing ppt benefit) then if no evidence, check that the application has means details (therefore not needing a ppt benefit)

I'm a little unsure what you mean by 'skipping' means passporting. If the means passporter was never run, then there would be no means passport. It being run again now would resolve to the same state. 

TBH, I've not put a huge amount of thought into this (order etc) because it needs revisiting soon anyway. The whole concept will quickly be nonsensical